content/docs/add-update-metadata-fields/_index.md

```diff
@@ -0,0 +1,180 @@
+
+At **Line 2**, `default=None` indicates the field is optional and by default undefined. Leave this line out for mandatory fields.
+
+At **Line 3**, we have a human-readable description of the field.
+
+At **Line 4**, we define the predicate. In this case, it's `dcterms:type`. Some common namespaces, like `DCTERMS` and `DCAT`, are imported by default. A full URI can also be defined, for example with `URIRef("http://example.com/range#property")`.
+
+At **Line 5**, we define the RDF type. There are many possible values here, such as `rdfs_literal`, `xsd:string`, or `uri`. It's recommended to take a look at other properties to understand what is necessary here.
+
+Once this is done, the JSON and YAML schemas need to be re-generated. For the `HRIDataset` class, this can be done by running the following command:
+
+```bash
+hatch run python sempyro/hri_dcat/hri_dataset
+```
+
+## FAIR Data Point
+
+For the technical point of view, updating the appropriate **SHACL shapes** allows for adding of fields.
+
+### Steps to Add a Field in FDP:
+
+1. In the FDP, log in as an admin user and go to the **Metadata schemas** option.
+2. Select the resource to update (e.g. **Catalog**).
+3. In the **Form Definition** textarea, add a new entry in the list of `sh:property` values. For example:
+
+```bash
+[
+  sh:path my:new-property ;      # the predicate IRI
+  sh:nodeKind sh:Literal ;       # the value type
+  sh:minCount 1 ;                # cardinality
+  dash:viewer dash:LiteralViewer ;  # UI hint for displaying
+  dash:editor dash:TextFieldEditor ; # UI hint for editing
+]
+```
+4. Click **Save** if this is a draft and needs further work, or **Save and release** if the work is done.
+5. Add a description and select a version number.
+6. Click **Release**.
+
+## Discovery Service
+
+The **Dataset Discovery** service requires two parts to be updated: the **OpenAPI definitions** and the **mapping**.
+
+### OpenAPI Definition
+
+Two definitions need to be updated, both located in the `src/main/openapi` folder: 
+- **ckan.yaml**: This file contains the API returned by CKAN. Based on this YAML, Java classes are automatically generated corresponding to the API definition. For adding a field to a **Dataset**, the primary change will likely be in the **CkanPackage** definition. See the examples there on how to add a property.
+- **discovery.yaml**: This file defines what the Discovery service should return. You can make this definition whatever you want it to be—it does not have to correspond one-to-one with CKAN. To add a property here, modify the **RetrievedDataset** definition. Again, see the examples in the file.
+
+### Mapping
+
+
+
+Once you have changed the definitions, follow these steps:
+1. Run the following command:
+
+```bash
+mvn clean compile
+```
```

Comment:
```suggestion
```
The command will probably generate a bunch of errors, but will regenerate the classes reflecting the OpenAPI objects

2. Add the mapping between the CKAN and Discovery service fields. The main place you want to look is most likely `src/main/java/io/github/genomicdatainfrastructure/discovery/utils/PackageShowMapper.java`, and modify the `RetrievedDatasetBuilder` See the code there on examples on how to map fields.
3. Update test cases, they are found in `src/test/java/io/github/genomicdatainfrastructure/discovery/services/PackageShowMapperTest.java`. Make sure to update 1. empty dataset examples 2. filled examples. You'll need to update both the `CkanPackage` objects (which reflects the CKAN API output) as well as the expected output, which is in the form of a `RetrievedDataset`.
4. Finally, test using both automatic testing `mvn test`, as well as run the package (`mvn compile quarkus:dev`) and check with Postman if mapping and output is as expected.
```

Reply:
This still needed to be written, if you can add it, thanks